test/test_torch.py

```diff
@@ -2926,56 +2926,37 @@ def test_tensor_randoms_bool(self):
         shapes = [(5, 0, 1), (0,), (0, 0, 1, 0, 2, 0, 0)]
         devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+        dtypes = [torch.bool, torch.float, torch.double, torch.half, torch.uint8,
+                  torch.int8, torch.short, torch.int, torch.long]
 
         for device in devices:
             for shape in shapes:
-                self.assertEqual(shape, torch.zeros(shape, device=device).shape)
-                self.assertEqual(shape, torch.zeros_like(torch.zeros(shape, device=device)).shape)
-                self.assertEqual(shape, torch.empty(shape, device=device).shape)
-                self.assertEqual(shape, torch.empty_like(torch.zeros(shape, device=device)).shape)
-                self.assertEqual(shape, torch.empty_strided(shape, (0,) * len(shape), device=device).shape)
-                self.assertEqual(shape, torch.full(shape, 3, device=device).shape)
-                self.assertEqual(shape, torch.full_like(torch.zeros(shape, device=device), 3).shape)
-                self.assertEqual(shape, torch.ones(shape, device=device).shape)
-                self.assertEqual(shape, torch.ones_like(torch.zeros(shape, device=device)).shape)
-                self.assertEqual(shape, torch.rand(shape, device=device).shape)
-                self.assertEqual(shape, torch.rand_like(torch.zeros(shape, device=device)).shape)
-                self.assertEqual(shape, torch.randn(shape, device=device).shape)
-                self.assertEqual(shape, torch.randn_like(torch.zeros(shape, device=device)).shape)
-                self.assertEqual(shape, torch.randint(6, shape, device=device).shape)
-                self.assertEqual(shape, torch.randint_like(torch.zeros(shape, device=device), 6).shape)
+                for dt in dtypes:
+                    self.assertEqual(shape, torch.zeros(shape, device=device, dtype=dt).shape)
+                    self.assertEqual(shape, torch.zeros_like(torch.zeros(shape, device=device, dtype=dt)).shape)
+                    self.assertEqual(shape, torch.full(shape, 3, device=device, dtype=dt).shape)
+                    self.assertEqual(shape, torch.full_like(torch.zeros(shape, device=device, dtype=dt), 3).shape)
+                    self.assertEqual(shape, torch.ones(shape, device=device, dtype=dt).shape)
+                    self.assertEqual(shape, torch.ones_like(torch.zeros(shape, device=device, dtype=dt)).shape)
+                    self.assertEqual(shape, torch.empty(shape, device=device, dtype=dt).shape)
+                    self.assertEqual(shape, torch.empty_like(torch.zeros(shape, device=device, dtype=dt)).shape)
+
+                    if (dt != torch.bool):
+                        self.assertEqual(shape, torch.empty_strided(shape, (0,) * len(shape), device=device, dtype=dt).shape)
+
+                    if (dt == torch.double or dt == torch.float or dt == torch.bool):
+                        self.assertEqual(shape, torch.rand(shape, device=device, dtype=dt).shape)
```

Comment:
OK, I'm not sure when this happened, but defining `torch.rand` for bool seems deeply questionable. `torch.rand` is defined to return random numbers in the uniform distribution between zero and one, inclusive and **exclusive**. This means that True should never be sampled; you should always get False. So it seems to me that we should **not** provide `torch.rand` for booleans; only randint.

Reply:
agreed, as mentioned in the previous PR, I think you only need to support randint and randint_like.